plugins/org.komodo.core/src/org/komodo/repository/ObjectImpl.java

```diff
@@ -1023,6 +1058,33 @@ private void setProperty( final Session session,
         }
     }
 
+    protected void setObjectProperty(UnitOfWork uow, String setterName, String propertyName,
+                                            Object value) throws KException {
+        UnitOfWork transaction = uow;
+
+        if (transaction == null) {
+            transaction = getRepository().createTransaction(getClass().getSimpleName() + HYPHEN + setterName, false, null); 
+        }
+
+        assert (transaction != null);
+
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(setterName + ": transaction = '{0}', value = '{1}'", //$NON-NLS-1$
+                         transaction.getName(),
+                         value);
+        }
+
+        try {
+            setProperty(transaction, KomodoLexicon.VdbModel.MODEL_DEFINITION, value);
```

Comment:
Maybe I'm missing something (sorry). I'll try again. In line 1078, shouldn't the 2nd argument be `propertyName` instead of the hard-coded property `KomodoLexicon.VdbModel.MODEL_DEFINITION`?